language/go/package.go

```diff
@@ -166,6 +169,54 @@ func (pkg *goPackage) inferImportPath(c *config.Config) error {
 	return nil
 }
 
+// Returns a a suitable go_library name based on the dir.
+// Major version suffixes (eg. "v1") are dropped.
+func libNameFromDir(dir string) string {
+	i := strings.LastIndexAny(dir, "/\\")
+	if i < 0 {
+		return dir
+	}
+	name := dir[i+1:]
+	if pkgVersionRe.MatchString(name) {
+		dir := dir[:i]
+		i = strings.LastIndexAny(dir, "/\\")
+		if i >= 0 {
+			name = dir[i+1:]
+		}
+	}
+	return name
+}
+
+// libNameByConvention returns a suitable lib name based on go_naming_convention
+// If go_default_library, "go_default_library" is returned.
+// Else if this is a 'main' package, 'foo_lib' is returned, where 'foo' is the name of the go_binary rule.
+// Else it is a regular package, the last segment of the importpath is returned. Major version suffixes (eg. "v1") are dropped.
+func libNameByConvention(nc namingConvention, binName, imp string) string {
+	if nc == goDefaultLibraryNC {
+		return defaultLibName
+	}
+	if binName != "" {
+		return binName + "_lib"
+	}
+	return libNameFromDir(imp)
+}
+
+// A simpler version of libNameByConvention for library dependencies.
+func resolvedLibName(nc namingConvention, imp string) string {
```

Comment:
Nit: probably just inline this function.

Reply:
Yeah you don't get a lot of bang from this tiny function. But I thought it would be nice to have all the naming convention code in a single place so you can easily see the semantics of the naming convention by reading a single place.

If you still want it inlined I can do it.